akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/PersistentConnection.scala

```diff
@@ -128,12 +139,32 @@ private[http2] object PersistentConnection {
             failStage(new RuntimeException(s"Connection failed after $maxAttempts attempts", cause))
           } else {
             setHandler(requestIn, Unconnected)
-            log.info(s"Connection attempt failed: ${cause.getMessage}. Trying to connect again${connectsLeft.map(n => s" ($n attempts left)").getOrElse("")}.")
-            connect(connectsLeft)
+            val nextEmbargo = embargo match {
+              case Duration.Zero => baseEmbargo
+              case otherValue    => (otherValue * 2).min(maxBaseEmbargo)
+            }
+            if (nextEmbargo == Duration.Zero) {
```

Comment:
This case is when the setting is disabled?

Reply:
Yep, I think perhaps the logic could be made a little bit simpler.